client2/commands.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: © 2023 David Stainton
+// SPDX-License-Identifier: AGPL-3.0-only
+package client2
+
+import sConstants "github.com/katzenpost/katzenpost/core/sphinx/constants"
+
+type Response struct {
+	// ID is the unique identifier with respect to the Request Payload.
+	// This is only used by the ARQ.
+	ID *[MessageIDLength]byte
+
+	// SURBID must be a unique identity for each request.
+	SURBID *[sConstants.SURBIDLength]byte `cbor:surbid`
+
+	// AppID must be a unique identity for the client application
+	// that is receiving this Response.
+	AppID uint64 `cbor:id`
+
+	// Payload contains the Response payload, a SURB reply.
+	Payload []byte `cbor:payload`
+
+	// IsStatus is set to true if Payload should be nil and IsConnected
+	// regarded as the latest connection status.
+	IsStatus bool `cbor:is_status`
+	// IsConnected is the latest connection status if IsStatus is true.
+	IsConnected bool `cbor:is_connected`
+
+	// IsPKIDoc is to to true if the Payload contains a PKI document stripped
+	// of signatures.
+	IsPKIDoc bool `cbor:is_pki_doc`
+}
+
+type Request struct {
+	// ID is the unique identifier with respect to the Payload.
+	// This is only used by the ARQ.
+	ID *[MessageIDLength]byte
+
+	// WithSURB indicates if the message should be sent with a SURB
+	// in the Sphinx payload.
+	WithSURB bool
+
+	// SURBID must be a unique identity for each request.
+	// This field should be nil if WithSURB is false.
+	SURBID *[sConstants.SURBIDLength]byte `cbor:surbid`
+
+	// AppID must be a unique identity for the client application
+	// that is sending this Request.
+	AppID uint64 `cbor:app_id`
```

Comment:
if it has to be unique we might as well make it a 128 bit byte array, then we don't have to worry about the threat of local applications bruteforcing each other's IDs.